app/uk/gov/hmrc/eoricommoncomponent/frontend/views/helpers/inputDate.scala.html

```diff
@@ -22,16 +22,16 @@
     form(id).errors ++ form(s"$id.day").errors ++ form(s"$id.month").errors ++ form(s"$id.year").errors
 }
 @errorContent(error: FormError) = {
-    <span id="@id-error" class="govuk-error-message govuk-!-margin-bottom-1">
+    <p id="@id-error" class="govuk-error-message govuk-!-margin-bottom-1">
         <span class="govuk-visually-hidden">Error: </span>
         @messages(error.message, error.args: _*)
-    </span>
+    </p>
 }
 
 @formattedErrors = {
   @if(formErrors.nonEmpty){
       @formErrors.headOption.map(err => messages(err.message, err.args: _*))
-      </span>
+      </p>
```

Comment:
@vigneshwaripandi where is this closing </p> tag being opened? Seems like that might be the cause of your problem. I think you can also reduce some of the boilerplate you have here by using more of the functionality provided by the RichDateInput(DateInput).withFormField (explanation of what it provides https://github.com/hmrc/play-frontend-hmrc#richdateinput) based on how you've got the errors in your form keyed on line 22 it should work out of the box for you

Though these errors will be the inline errors for the field, if you want to display an error summary, you'd use the GovukErrorSummary component https://design-system.service.gov.uk/components/error-summary/ and there is a helper for that too that makes integration with play a bit easier https://github.com/hmrc/play-frontend-hmrc#richerrorsummary